app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -1450,15 +1456,8 @@ DeviceDelegateOpenXR::UpdateHandMesh(const uint32_t aControllerIndex, const std:
   if (!m.handMeshRenderer)
     return;
 
-  if (m.handMeshMSFT.isSupported) {
-    HandMeshBufferPtr buffer = m.input->GetHandMeshBuffer(aControllerIndex);
-    if (buffer) {
-      m.handMeshRenderer->Update(aControllerIndex, handJointTransforms, aRoot, buffer, aEnabled, leftHanded);
-      return;
-    }
-  }
-
-  m.handMeshRenderer->Update(aControllerIndex, handJointTransforms, aRoot, aEnabled, leftHanded);
+  HandMeshBufferPtr buffer = m.input->GetNextHandMeshBuffer(aControllerIndex);
+  m.handMeshRenderer->Update(aControllerIndex, handJointTransforms, aRoot, buffer, aEnabled, leftHanded);
```

Comment:
This looks dangerous as `GetNextHandMeshBuffer` might return null. There was a null check in the previous version of the code

Reply:
We need to call Update() here regardless (to hide the hands if aEnabled is false), but since now we have just one version of update, we pass buffer here as-is without checking null-ness.

Inside the Update() method in each strategy, we handle the null-ness of buffer. For all strategies other than Geometry, we actually assert that buffer is null since it is not needed there. For Geometry, we already bail early if buffer is null.

Reply:
Beh, sorry for the noise, somehow I was assuming you're calling buffer->Update :facepalm: Forget about it